packages/flutter_tools/lib/src/commands/attach.dart

```diff
@@ -218,8 +218,7 @@ class AttachCommand extends FlutterCommand {
         FuchsiaIsolateDiscoveryProtocol isolateDiscoveryProtocol;
         try {
           isolateDiscoveryProtocol = device.getIsolateDiscoveryProtocol(module);
-          observatoryUri = await isolateDiscoveryProtocol.uri;
-          printStatus('Done.'); // FYI, this message is used as a sentinel in tests.
+          observatoryUri = Stream<Uri>.fromFuture(isolateDiscoveryProtocol.uri).asBroadcastStream();
```

Comment:
Here and elsewhere, don't use broadcast streams. These do not buffer events, so if the future resolves before the first subscriber is added the event will be lost

Reply:
This seems to work fine:

```dart
import 'dart:async';

void main() async {
  Stream<String> stream = Stream<String>.fromFuture(Future<String>.value('foo'))
      .asBroadcastStream();

  stream.listen((String value) {
    print(value);
  });
}

```
prints `foo`.

Reply:
The docs for `asBroadcastStream()` [here](https://api.dartlang.org/stable/2.6.1/dart-async/Stream/asBroadcastStream.html):

> The returned stream will subscribe to this stream when its first subscriber is added, and will stay subscribed until this stream ends, or a callback cancels the subscription.

In general though, I don't know if it would be needed to pass `onListen` and `onCancel` callbacks to `asBroadcastStream()` to pause and resume the underlying stream when the number of subscribers hits zero.